### ISSUE TITLE
'Top-Level trusted' sometimes still does not work and breaks websites until manual (ctrl-f5) reload

### ISSUE THREAD

**Issue by tokariu** [hackademix/noscript#408]
FF v133.0.3 / NS v11.5.2

Having checked 'Top-Level trusted' in the settings still does not work every now and then.
then it breaks the site due to not activating js for the current visited top-level-domain. a ctrl-f5 reload fixes the bug most of the time for the visited site.
this happens on random sites and not every time, but on a pretty regular basis.
youtube is broken since month' (1st video click might work, the 2nd doesn't)
this error is not just tied to youtube, it happens on any website - just sometimes. 

I just found that there was an issue before, so this is a kind of a continuation of it: https://github.com/hackademix/noscript/issues/388

**Comment by hackademix** [comment]
Can you see this happening on [12.0.901](https://noscript.net/getit/#devel)?

**Comment by tokariu** [comment]
> Can you see this happening on [12.0.901](https://noscript.net/getit/#devel)?

installed, reporting back after testing for a while.

**Comment by tokariu** [comment]
after a few days of testing, I recognized no problems with the exception of youtube - which is still broken (youtube -> search video -> click on any video -> it gets stuck loading, the pixel-progress bar on top of the site is stuck at approx 75%) it's working if I uninstall the noscript addon.

**Comment by tokariu** [comment]
I notice that visiting imdb.com and clicking some movie details is crashing the FF tab when NoScript is enabled. It shows me a tab-has-crashed window and I have to reload the tab. in the 2nd attempt it is working for that particular page. Clicking other links or movies reintroduce the issue again.

**Comment by hackademix** [comment]
Could you please:

1. Manually upgrade to [12.1.901](https://noscript.net/getit/#devel)
2. Check whether the 2 problems persist
3. If they do, share your _NoScript Options>Settings>Export_ (privately, if you prefer, to the email on the top right of https://maone.net)

? Thanks!

**Comment by tokariu** [comment]
alright, will test and report again.

**Comment by humanzgnome** [comment]
Regarding "Temporarily set top-level sites to TRUSTED" intermittently not working:
I have version 12.1.1.  The bug occurs on the first loading from a particular domain name.  I use tinkercad.com to demonstrate. This page relies on script to load nearly everything displayed, the script is blocked and loading hangs with just a "busy" icon displayed. Any subsequent use of the domain, in my example tinkercad.com, works as expected. So, apparently, the top level is being set to TRUSTED - just not in time for that first page load. One must restart the browser to see the behavior again. 
Here are steps to recreate behavior:
concisely:
enable "Temporarily set top-level sites to TRUSTED"
close the browser, restart the browser
navigate to tinkercad.com, loading hangs
click reload button, loading succeeds
navigate to tinkercad.com/things, loading succeeds

repeat and see same behavior"
close the browser, restart the browser
navigate to tinkercad.com, loading hangs
click reload button, loading succeeds
navigate to tinkercad.com/things, loading succeeds

now reverse the page load order and see first (different) load from domain hang
close browser, restart the browser
navigate to tinkercad.com/things, loading hangs
navigate to tinkercad.com, loading succeeds

TLDR:
exactly what I did (used new profile with noscript and no other changes)
add new profile
continue without account
add a name (to the profile), I used "zzz"
extensions/visit chrome store
search for  noscript
add (noscript) to chrome. Version is 12.1.1
close "turn on sync" with the upper right X
pin it to toolbar
view noscript options by clicking the "wrench" icon
enable "Temporarily set top-level sites to TRUSTED" by "checking" the box
close the "noscript settings" tab 
close the browser
restart the browser
clicking the user icon in the upper right, select the newly created profile, mine is "zzz"

navigate to tinkercad.com, loading hangs
click reload button, loading succeeds
navigate to tinkercad.com/things, loading succeeds

repeat (and see same behavior):
close browser, restart the browser
select the newly created profile, mine is "zzz"

navigate to tinkercad.com, loading hangs
click reload button, loading succeeds
navigate to tinkercad.com/things, loading succeeds

now reverse the page load order
close browser, restart the browser
select the newly created profile, mine is "zzz"

navigate to tinkercad.com/things, loading hangs
navigate to tinkercad.com, loading succeeds

**Comment by tokariu** [comment]
> alright, will test and report again.

@hackademix regarding my previous testings.. it seems like google changed the way how things work at youtube.

before the issue, I was able to run youtube videos fine with just "Top-Level trusted" so that only youtube.com scripts are allowed.
it seems youtube also REQUIRES now googlevideo.com, google.com and gstatic.com to be whitelisted as trusted to be able to work.
(I know some of these are whitelisted by default in noscript when you install noscript fresh from scratch, but I decided to delete all whitelistings before so I have full control and I prefer this more privacy oriented approach.)

so if those 3 are not whitelisted youtube videos will play only for ~20 seconds and then just stop. Skipping in the timeline with the seeking bar does not work either.

It works fine if you whitelist them all to "temporarily trusted" or "trusted" as seen in this pic:
![Image](https://github.com/user-attachments/assets/9735fcad-ffe0-4a16-8582-b551759742fe)

However, I don't like giving these 3 domains general whitelistings for all websites. So I leave them untrusted and I have to whitelist all 3 of them EACH time I want to watch a youtube video.

As said: previously, before I noticed the issue, youtube videos worked just fine with only youtube.com whitelisted. so I guess youtube changed something in their backend that requires the other sites now. 

I wish there was a way to fix this without whitelisting all the other domains.
